mmv1/products/netapp/volume.yaml

```diff
@@ -0,0 +1,416 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'Volume'
+description: |
+  A volume is a file system container in a storage pool that stores application, database, and user data.
+
+  You can create a volume's capacity using the available capacity in the storage pool and you can define and resize the capacity without disruption to any processes.
+
+  Storage pool settings apply to the volumes contained within them automatically.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Quickstart': 'https://cloud.google.com/netapp/volumes/docs/get-started/quickstarts/create-volume'
+    'Documentation': 'https://cloud.google.com/netapp/volumes/docs/configure-and-use/volumes/overview'
+  api: 'https://cloud.google.com/netapp/volumes/docs/reference/rest/v1/projects.locations.volumes'
+base_url: projects/{{project}}/locations/{{location}}/volumes
+self_link: projects/{{project}}/locations/{{location}}/volumes/{{name}}
+create_url: projects/{{project}}/locations/{{location}}/volumes?volumeId={{name}}
+update_url: projects/{{project}}/locations/{{location}}/volumes/{{name}}
+update_verb: :PATCH
+update_mask: true
```

Comment:
We still want an update test to make sure that this works, even if it's only for labels and description

Reply:
HI @slevenick, thank you for your review. They all touch the volume part, which have just been included make the snapshot resource work. The volume resource itself is handled in https://github.com/GoogleCloudPlatform/magic-modules/pull/9513 and was merged yesterday (after snapshot was opened). So please ignore them for the snapshot review.

Having said that, let me comment on your individual feedbacks:
The update test is testing labels and description, see https://github.com/GoogleCloudPlatform/magic-modules/blob/c1faf0477cb1c17a92e1a14ed080408dd011c022/mmv1/third_party/terraform/services/netapp/resource_netapp_volume_test.go#L132.